peci: driver: Ping command failing due to incorrect tx length handling and wait timeout

DiwakarCC opened the pull request:
**Bug description**

Below 2 issues has to be fixed:

Issue 1: Ping command have write and read length of '0'. In current implementation, when ping command is sent to driver, payload of size (Tx_Buf_Len-1) is added to the FIFO through below loop.

for (int i=0; i<(Tx_Buf_Len-1); i))
{
// Add to FIFO
}

This works for all PECI commands except for ping command. Incase of ping command, Tx_Buf_Len=0 results in comparison (0 < -1(0xFFFFFFFF)) which is comparison b/w signed and unsigned numbers. When signed (index i) and unsigned (Tx_Buf_Len) compared, compiler implicitly converts signed to unsigned results in entering infinite loop (since 0<0xFFFFFFFF is true).

Issue 2: wait_timeout never decrements hence the loop is waiting infinitely for (!(base->STATUS1 & MCHP_PECI_STS1_EOF)) to become true. 

**Steps to reproduce:**
Send peci_ping() command to peci driver.

**Expected behavior:**
Ping command tx should success.

**Impact:**
Peci app will fail for ping command.

nashif committed:
toolchain: make clang happy with inline asm

clang did not like %p on x86, make it use %c.

Did not explore why this is happening..

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
clang: do not assert on libgcc not found

We do not need libgcc always, some environments do not have libgcc and
do not require it, so keep it more flexible.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nandojve committed:
samples: net: cloud: Introduce TagoIO IoT cloud http post

Introduce TagoIO IoT Cloud HTTP post client example.  This explorer
Zephyr network resources to demonstrate an end to end application.
The TagoIO allows that any user can test on a easy way Ethernet, WIFI
and Modem (PPP) with BSD sockets.  The example provides overlays to
configure WIFI and Modem.

The application consists an a pseudo temperature sensor that sends
periodically data to TagoIO IoT Cloud platform.  The data can be
visualized on a web browser dashboard, cellphone or tablet.  The
steps to configure TagoIO are described on the example documentation.

Special Variables:
 - CONFIG_TAGOIO_DEVICE_TOKEN   DEVID  token generated by TagoIO
 - CONFIG_TAGOIO_HTTP_WIFI_SSID SSID   when using WIFI
 - CONFIG_TAGOIO_HTTP_WIFI_PSK  PASSWD when using WIFI
 - CONFIG_MODEM_GSM_UART_NAME   UART   label when using MODEM
 - CONFIG_MODEM_GSM_APN         APN    when using MODEM

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nashif committed:
boards: qemu_x86: enable llvm toolchain

Make this board buildable with llvm.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

Commit:
samples: civetweb_websocket_server: init project

initialize this project by adding corresponding project files

Signed-off-by: Alexander Kozhinov <AlexanderKozhinov@yandex.com>

de-nordic committed:
subsys/fs: Make FAT FS formatting code optional

The commit adds FS_FAFTFS_MKFS Kconfig option; the option, y by default,
allows to include mkfs supporting code, within fs_mount callback,
for ELM FAT file system driver.

When formatting of FAT FS volumes is not desired, this option may be
deselected to slightly reduce code size.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

carlocaione committed:
arch: arm64: Add missing vector table entries

The current vector table is missing some (not used) entries. Fill these
in for the sake of completeness.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

carlocaione committed:
arch: arm64: Use mov_imm when possible in the start code

Instead of relying on mov.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

rlubos committed:
net: lwm2m: Notify the application on network error

Add a simple backoff mechanism between consecutive registration attempts
in case of registration failures. Finally, notify the application in
case the registration failed several times.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
net: lwm2m: Move bootstrap registration send into a separate function

Refactor the boostrap regstration procedure, by splitting the message
creation and sending into a separate function, in similar manner as
it's done with regular registration.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

carlocaione committed:
aarch64: mmu: Move MMU files in a sub-directory

We are probably going to do more work on the MMU side and more files
will be added. Create a new sub-directory to host all the MMU related
files.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

carlocaione committed:
aarch64: mmu: Create new header file

The MMU coded is polluted by a lot of preprocessor code. Move that code
to a proper header file.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

carlocaione committed:
aarch64: mmu: Enable support for unprivileged EL0

The current MMU code is assuming that both kernel and threads are both
running in EL1, not supporting EL0. Extend the support to EL0 by adding
the missing attribute to mirror the access / execute permissions to EL0.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

carlocaione committed:
arch: arm64: Reword comments

Fix, reword and rework comments.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

carlocaione committed:
arch: arm64: Fix indentation

Fix indentation for the ISR wrapper.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

carlocaione committed:
arch: arm64: Rename z_arm64_svc to z_arm64_sync_exc

The SVC handler is not only used for the SVC call but in general for all
the synchronous exceptions. Reflect this in the handler name.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

carlocaione committed:
arch: arm64: Remove _BIT suffix

This is redundant and not coherent with the rest of the file. Thus
remove the _BIT suffix from the bit field names.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

andrzej-kaczmarek committed:
Bluetooth: controller: Remove redundant check

We handle secondary channel only for extended advertising sets, that
means we always have ADV_EXT_IND on primary channel so there is no need
to support non-ext PDUs here.

Signed-off-by: Andrzej Kaczmarek <andrzej.kaczmarek@codecoup.pl>

Commit:
x86: define some unused PTE bits

These are ignored by the CPU and may be used for OS
accounting.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
x86: use unused PTE bits when mapping memory

Page table management for x86 is being revised such that there
will not in many cases be a pristine, master set of page tables.
Instead, when mapping memory, use unused PTE bits to store the
original RW, US, and XD settings when the mapping was made.

This will allow memory domains to alter page tables while still
being able to restore the original mapping permissions.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

laurenmurphyx64 committed:
boards: Add qemu_x86_nokpti target

Adds a build target to run tests on qemu_x86 with KPTI disabled.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

laurenmurphyx64 committed:
boards: Add qemu_x86_64_nokpti target

Adds a build target to run tests on qemu_x86_64 with KPTI disabled.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

Commit:
x86: don't reserve room for page tables in stack

These are handled at the memory domain level now.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

erwango committed:
drivers/pinmux: stm32: Deprecate PWM related macros

Following migration of all in-tree boards to device tree bindings
for pwm pins configuration, deprecate PWN related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

carlocaione committed:
arch: arm64: Remove unused macro parameters

Remove z_arm64_{enter,exit}_exc parameter leftovers.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

de-nordic committed:
subsys/fs: Make FAT FS write support optional

The commit adds FS_FATFS_READ_ONLY Kconfig option; the option, when
selected, excludes write supporting code within ELM FAT driver.

When write support to FAT FS volumes is not desired, this option may be
selected to slightly reduce code size.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

mbolivar-nordic committed:
doc: devicetree: reorganize a bit

Rework some section titles and separate the API into its own sub-page
in the reference section.

This is prep work for adding generated reference material on
devicetree bindings.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

de-nordic committed:
fs: Add support for mount flags

The flags field has been added to fs_mount_t structure, accompanied
with two new flags:
  FS_MOUNT_FLAG_READ_ONLY -- mount fs as read only
  FS_MOUNT_FLAG_NO_FORMAT -- do not format volume when system not found

Code supporting the flags has been added to FS layer and drivers for
LittleFS and FAT FS.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

erwango committed:
drivers/pinmux: stm32: Deprecate (LP)U(S)ART related macros

Following migration of all in-tree boards to device tree bindings
for serial pins configuration, deprecate (LP)U(S)ART related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

nandojve committed:
runners: bossac: Add speed argument

The current stty command uses a hard code value of 1200.  This is not
compliant with SAM-BA specs and may create compatibility problems.  Add
an optional speed argument with 115200 as default value following SAM-BA
specifications.  All boards that needs a different speed should define
board_runner_args(bossac "--speed=<value>") with value as required
speed.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

gmarull committed:
runners: add support for stm32cubeprogrammer

Add support for the official ST Microelectronics programming tool (CLI
version).

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
boards: arm: nucleo_h743zi: add support for stm32cubeprogrammer runner

Add support for the recently introduced STM32CubeProgrammer runner.
Updated documentation to mention its availability as latest official
OpenOCD releases do not support H7 series.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
runners: stm32cubeprogrammer: add tests

Add tests for the stm32cubeprogrammer runner.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

andrzej-kaczmarek committed:
Bluetooth: controller: Update HCI supported cmds with ext adv

Return proper bitmask of supported commands when extended advertising
enabled.

Signed-off-by: Andrzej Kaczmarek <andrzej.kaczmarek@codecoup.pl>

erwango committed:
drivers/pinmux: stm32: Deprecate CAN related macros

Following migration of all in-tree boards to device tree bindings
for serial pins configuration, deprecate CAN related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/pinmux: stm32: Deprecate DAC related macros

Following migration of all in-tree boards to device tree bindings
for CAN pins configuration, deprecate DAC related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

mbolivar-nordic committed:
doc: reference: add devicetree bindings content

Add generated documentation content for known devicetree bindings
using the Binding abstraction which was just added to edtlib. This
works similarly to the way Kconfig content is generated, so extract a
bit of common helper code for doing that out and rename the relevant
files to keep the distinction clear.

Make the documentation build system respect a preset DTS_ROOT. In this
way, out of tree bindings can be added to the generated content by
telling the documentation build system where to find them, identically
to how out of tree bindings can be added to a Zephyr application.

Similarly, make the output directory configurable.

Fixes: #28865
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

rlubos committed:
net: lwm2m: Add lwm2m_send_empty_ack() function to internal API

So far this function existed as a static function in LwM2M PULL FOTA
module. Since such functionality will be needed in other places, make it
an internal API function.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
net: lwm2m: Fix separate response handling

Separate response handling implemented in the engine was faulty. The
separate response was not acknowledged by the client, resulting in
spurious retransmissions from the server side.

Also, the pending CON message was retransmitted by the client even after
it was acknowledged by an empty ACK, but the respnse haven't arrived
yet. Fix this by adding a new `acknowledged` flag to the `lwm2m_message`
structure. Once acknowledged, the flag is set and the confirmable
message is no longer retransmitted. We keep the message on the pending
list in order to timeout properly in case separate response does not
arrive in time.

Finally, prevent the reply callback from being called twice in case
the response is transmitted separately from ACk. The callback should
only be called on the actual reply, not the empty ACK.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
net: lwm2m: Remove handle_separate_response flag

It shouldn't be optional to handle separate response, as it's a
mandatory requirement according to the RFC7252:

"The protocol leaves the decision whether to
 piggyback a response or not (i.e., send a separate response) to
 the server.  The client MUST be prepared to receive either."

Therefore, remove the flag as separate responses are handled now
properly.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

Commit:
tests: userspace: fix tests that don't hold on MMU

MMU threads within the same memory domain have access to
each other's stacks.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
x86: smp: add TLB shootdown logic

This will be needed when we support memory un-mapping, or
the same user mode page tables on multiple CPUs. Neither
are implemented yet.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

gmarull committed:
drivers: pwm: stm32: fix timer clock calculation

Calculation of the timer clock was wrong for some F4/F7/H7 series.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

Commit:
review: rb: develop comments and add some details

Add some comments about rbtree to make it more readable
Add more detail infos to make the purpose and process
of the test cases more clear which include test goal,
test step, input, judging criteria, constraints, etc.,
and these can be seen in our Zephyr documentations.

Signed-off-by: Ningx Zhao <ningx.zhao@intel.com>

de-nordic committed:
tests/fs/fs_api: Add API tests fo FS_MOUNT_FLAG_READ_ONLY

The commit adds read-only mount scenario tests to fs_api test
suite.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
tests/fs/littlefs: Add tests for FS_MOUNT_FLAG_*

Adds various tests for mounting LittleFS with FS_MOUNT_FLAG_NO_FORMAT
and FS_MOUNT_FLAG_READ_ONLY, and operations on read-only mounted
file system.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
tests/fs/fatfs: Add tests for FS_MOUNT_FLAG_*

Adds various tests for mounting ELM FAT FS with FS_MOUNT_FLAG_NO_FORMAT
and FS_MOUNT_FLAG_READ_ONLY, and operations on read-only mounted
file system.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

pabigot committed:
tests: kernel: work_queue_api: relax test to eliminate racy checks

We can't control ticks accurately enough to detect the transition
between on a queue and being handled, so relax the checks to make
things pass.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

mbolivar-nordic committed:
dts: update vendor-prefixes.txt

Make the formatting consistent and add a missing prefix we are using.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
scripts: edtlib: bindings_from_paths() helper function

This is a convenience function for creating a bunch of Binding objects
from files in a directory.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

nashif committed:
toolchain: clang: add -Wno-typedef-redefinition option

Without the -Wno-typedef-redefinition option, clang complains if a
typedef gets redefined in gnu99  mode (since this is officially a C11
feature).

While new versions of GCC do not seem to issue this warning in gnu99
mode anymore. So some existing code with typedef redefined which works
well with GCC will issue this warning.

Similar to what was done in 2354f055ec41fa49f8fde5f5ffca0f84313b7544.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

Abhishek-brcm committed:
drivers: pcie_ep: iproc: clear pvm interrupts before handling

We should clear the pvm interrupts (snoop and pcie pmon lite interrupt)
at source before handling them.

This will make sure that we do not lose any interrupts that may have
been asserted to interrupt controller during the handling routine.

Signed-off-by: Abhishek Shah <abhishek.shah@broadcom.com>

Commit:
userspace: move mem_domain->arch first

This sub-struct may have alignment requirements, move it to
the first member.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
x86: move page tables to the memory domain level

- z_x86_userspace_enter() for both 32-bit and 64-bit now
  call into C code to clear the stack buffer and set the
  US bits in the page tables for the memory range.

- Page tables are now associated with memory domains,
  instead of having separate page tables per thread.
  A spinlock protects write access to these page tables,
  and read/write access to the list of active page
  tables.

- arch_mem_domain_init() implemented, allocating and
  copying page tables from the boot page tables.

- struct arch_mem_domain defined for x86. It has
  a page table link and also a list node for iterating
  over them.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

YuguoWH committed:
arch: arc: fix a reg misuse in leaving tickless idle

There is a register misuse in leaving tickless idle code, which would
destroy exception/interrupt status. This commit fix this issue.

Signed-off-by: Yuguo Zou <yuguo.zou@synopsys.com>

YuguoWH committed:
arch: arc: use ifdef to replace if define in isr wrapper code

isr wrapper code has mixed usage of #ifdef and #if define macros. Unify
them to more usual #ifdef.

Signed-off-by: Yuguo Zou <yuguo.zou@synopsys.com>

de-nordic committed:
west.yml: Snapshot update of mcumgr from upstream

This commit updates mcumgr with latest snapshot from the upstream.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

Thalley committed:
Bluetooth: host: Handle PA sync cancel event

The sync established event caused by a cancel by the host
was not properly handled. When cancelling the sync established
event is created, and the sync object should not be deleted
before that event is received.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

de-nordic committed:
fs: Make fs API callback runt-time check optional

The commit changes run-time checks of file system driver's callbacks
pointers, against null, optional by replacing `if` with `CHECKIF` macro,
which means that they can be removed from compilation with Kconfig
option CONFIG_NO_RUNTIME_CHECKS.
Additionally the commit allows the same checks to fail hard, with
assertion, when CONFIG_ASSERT_ON_ERRORS option is selected.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

Thalley committed:
Bluetooth: shell: Avoid registering PA sync callbacks more than once

The PA sync callbacks would have been registered for each new
PA sync, which would cause the callbacks to be called multiple times
if multiple PA syncs were created.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

ioannisg committed:
arch: introduce option to force internal architectural state init

We introduce an option that instructs Zephyr to perform
the initialization of internal architectural state (e.g.
ARCH-level HW registers and system control blocks) during
early boot to the reset values. The option is available
to the application developer but shall depend on whether
the architecture supports the functionality.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
arch: arm: cortex_m: implement functionality for ARCH core regs init

Implement the functionality for configuring the
architecture core registers to their warm reset
values upon system initialization. We enable the
support of the feature in the Cortex-M architecture.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
zephyr: make images loaded by MCUboot enable arch core HW regs init

Instruct Zephyr images that are loaded by MCUboot to
force the initialization of architecture core HW registers,
if the architecture supports the functionality.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

erwango committed:
drivers/pinmux: stm32: Deprecate I2C related macros

Following migration of all in-tree boards to device tree bindings
for I2C pins configuration, deprecate I2C related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/pinmux: stm32: Deprecate SPI related macros

Following migration of all in-tree boards to device tree bindings
for SPI pins configuration, deprecate SPI related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/pinmux: stm32: Deprecate ADC related macros

Following migration of all in-tree boards to device tree bindings
for ADC pins configuration, deprecate ADC related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/pinmux: stm32: Deprecate ETH related macros

Following migration of all in-tree boards to device tree bindings
for ETH pins configuration, deprecate ETH related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: stm32: Move USB pins to device tree configuration

Move STM32 based board USB pin configuration to device tree.

Exceptions:
* olimex_stm32_h407: Node not enabled and not documented.
Signal added in disabled node.
* L0/G4 based boards as signals are not available yet.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

cfriedt committed:
drivers: ieee802154: cc13xx_cc26xx: use ti rf driver api

This change reworks the cc13xx_cc26xx IEEE 802.15.4 driver to use
the TI RF driver API that is available in modules/hal/ti.

There are a number of benefits to using TI's API including
 - a stable multi-OS vendor library and API
 - API compatibility with the rest of the SimpleLink SDK and SoC family
 - potential multi-protocol & multi-client radio operation
   (e.g. both 15.4 and BLE)
 - coexistence support with other chipsets via gpio
 - vetted TI RF driver resources, such as
   - the radio command queue
   - highly tuned / coupled RTC & RAT (RAdio Timer) API

Fixes #26312

Signed-off-by: Christopher Friedt <chrisfriedt@gmail.com>

MaureenHelm committed:
disk: Fix nxp usdhc driver optional gpio properties

The power and chip detect gpios are optional device tree properties for
the nxp usdhc driver. Fixes a build error on the mm_swiftio board, which
does not set the power gpio property.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
boards: samples: Enable fat_fs sdhc sample on mm_swiftio board

Enables the fat_fs sample on the mm_swiftio board by adding a
board-specific Kconfig overlay, and adding sdhc to the list of
board-supported features.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

Commit:
Tests: Minor change in description and group of SMP test cases

Do minor change of the descriptions and doxygen group name in order to
pave the way for generation the test specification.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

erwango committed:
BK: Tmp increase number of jobs (DNM)

Only to pass CI


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

cfriedt committed:
ieee802154: cc13xx_cc26xx: workaround for issue in ti driver

This is a temporary workaround for an issue in TI's RF Driver
API. A subsequent release of the SimpleLink SDK will mitigate
the need for it and it can be reverted when hal/ti receives
that update.

Fixes #29418

Signed-off-by: Christopher Friedt <chrisfriedt@gmail.com>

Commit:
tests: mem_protect: add migration test case

We need to make sure that if we migrate a thread to another
memory domain, the migration process doesn't cause the target
thread to explode. This is mostly a concern on SMP systems;
the thread could be running on another CPU at the same time.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

nandojve committed:
drivers: wifi: eswifi: Drop global data visibility

Remove global access to structure eswifi_spi_data variable.  Instead,
add a method to pass access to that structure.  This allows better
control to the data bus variable.

Signed-off-by: Gerson Fernando Budke <gerson.budke@atl-electronics.com>

nandojve committed:
drivers: wifi: eswifi: Add uart bus interface

Add uart bus interface to extended esWIFI driver.  This enables all
Inventek modules with IWIN AT Commands firmware.

Signed-off-by: Gerson Fernando Budke <gerson.budke@atl-electronics.com>

jukkar committed:
net: tcp2: Avoid deadlock when closing the connection

When a connection is being closed, it is possible that the application
will have a lock to net_context and TCP2 connection lock. If we then
receive a final TCP2 ACK and close the connection, the locking order
get switched and TCP2 will first try to get its own lock and then the
net_context lock. This will lead to deadlock as the locking ordering
is now mixed.

The solution is to unref the TCP connection after releasing the
connection lock. The TCP connection unref function will anyway get the
lock so no need to do double locking.

Fixes #29444

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

nandojve committed:
boards: shields: Introduce inventek es-WIFI shield

Add Inventek es-WIFI modules shield.  This shield exposes es-WIFI driver
using Arduino Uno R3 header by UART or SPI interfaces.  It shows how
user can create their own overlay and expose es-WIFI driver.

The current Inventek's EVB doesn't have all pins necessary to control
the module by Arduino hearder.  This shows how to wire to get ISM43xx
EVB working.

Signed-off-by: Gerson Fernando Budke <gerson.budke@atl-electronics.com>

rlubos committed:
samples: net: lwm2m_client: Add bootstrap overlay file

Add overlay-bootstrap.conf which allows to easily configure the sample
to use the bootstrap server.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

Commit:
x86: move page table reservation macros

We don't need this for stacks any more and only use this
for pre-calculating the boot page tables size. Move to C
code, this doesn't need to be in headers anywhere.

Names adjusted for conciseness.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
boards: x86: tune X86_MMU_PAGE_POOL_PAGES

These are set such that we have enough pages in the pool
for typical driver mappings and to instantiate two more
memory domains, which is what our tests require.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
x86: update X86_MMU_PAGE_POOL_PAGES documentation

Help users understand how this should be tuned. Rather than
guessing wildly, set the default to 0. This needs to be tuned
on a per-board, per-application basis anyway.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

MaureenHelm committed:
drivers: ethernet: Rework mcux driver to use DT_INST_FOREACH

Reworks the mcux ethernet driver to use DT_INST_FOREACH_STATUS_OKAY and
eliminate a lot of duplicate code between instance 0 and instance 1.

Renames the ERR_MISC interrupt to ERR due to an issue with the
underscore in the device tree preprocessor macros.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

Commit:
x86: add logs for tuning page pool size

This will do until we can set up a proper page pool using
all unused ram for paging structures, heaps, and anonymous
mappings.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

nandojve committed:
samples: net: wifi: Enable inventek es-WIFI shield tests

The es-WIFI driver is used by Inventek's es-WIFI shields. This enables
wifi sample to build all variations to ensures that dependencies are
meet.

Signed-off-by: Gerson Fernando Budke <gerson.budke@atl-electronics.com>

eHammarstrom committed:
manifest: sync mbedtls alpn extension

- Adds mbedtls alpn negotiation support

Signed-off-by: Emil Hammarstrom <emil.a.hammarstrom@gmail.com>

rlubos committed:
samples: net: lwm2m_client: Update docs with bootstrap information

Add bootstrap section to the documentation, explaining how to configure
the Leshan Demo Bootstrap Server locally and run the sample.

Update the download link for the Leshan Demo Server, according to the
upstream Leshan documentation.

Fix bullet list with Leshan security parameters.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
samples: net: lwm2m_client: Increase log_strdup buffer count

The sample was producing quite a few `<log_strdup alloc failed>`
warnings when running in default configuration on qemu_x86. Increase the
log_strdup buffer count to prevent that.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

Commit:
x86: add qemu_x86_tiny

This build target has all the low-memory options enabled for
memory management: a 4MB address space, 32-bit paging mode,
no KPTI, an empty page pool, and common page tables for
memory domains.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

trond-snekvik committed:
sys: util: Add clamp macro

Adds CLAMP macro to complement the current min/max macros, as well as a
gcc specific Z_CLAMP macro for single-evaluation expansion.

CLAMP combines the functionality of MIN and MAX, eliminating the
bug-prone usage of MIN(MAX(value, FLOOR), CEIL) found throughout the
codebase in every possible combination.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

trond-snekvik committed:
sys: util: Replace MIN(MAX(a, b), c) with CLAMP

Replaces all existing variants of value clamping with the MIN and MAX
macros with the CLAMP macro.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

rlubos committed:
net: lwm2m: Fix PULL FW update in case of URI parse errors

The memset on firmware_ctx during PULL FW update initialization will
set the socket descriptor to a valid value of 0. This leads to an error
if parsing of the URI provided by the server fails, and the firware_ctx
is closed - the socket with a descriptor 0 will be accidently closed.
Fix this by invalidating the socket FD after the memset on
initialization.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

de-nordic committed:
fs: Document fs_open and fs_seek flags

Descriptions of fs_open flags have been added and descriptions of
whence options to_fs_seek have been moved, to make them visible
in html documentation.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

ioannisg committed:
doc: mention the INIT_ARCH_HW_AT_BOOT functionality in the rel notes

Mention the newly introduced funtionality for Cortex-M
architecture to force the early-boot initialization of
internal architecture HW state.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

Commit:
x86: add support for common page tables

We provide an option for low-memory systems to use a single set
of page tables for all threads. This is only supported if
KPTI and SMP are disabled. This configuration saves a considerable
amount of RAM, especially if multiple memory domains are used,
at a cost of context switching overhead.

Some caching techniques are used to reduce the amount of context
switch updates; the page tables aren't updated if switching to
a supervisor thread, and the page table configuration of the last
user thread switched in is cached.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
x86_64: disable shared_mem test until bugs fixed

re-enabling this test was premature, it still crashes very often
due to two known issues affecting x86 64-bit. Filter out for
every 64-bit x86 platform until those bugs are solved, which
will reduce CI failure noise.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
userspace: update k_mem_domain_add_partition docs

We no longer require partitions to be within RAM bounds,
MMIO or other mapped memory are fine too.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

andrzej-kaczmarek committed:
Bluetooth: controller: Remove redundant check

ull_adv_data_set() is only used for legacy advertising data (either via
legacy HCI command or as fallback for extended advertising set which
uses legacy PDUs) so there is no need to check for extended advertising.

Signed-off-by: Andrzej Kaczmarek <andrzej.kaczmarek@codecoup.pl>

andrzej-kaczmarek committed:
Bluetooth: controller: Verify AD length for legacy adv

We should not allow setting more than 31 bytes of AD on legacy
instances.

Signed-off-by: Andrzej Kaczmarek <andrzej.kaczmarek@codecoup.pl>

rlubos committed:
net: lwm2m: Send Registration Update on lifetime change

According to LwM2M specfication v1.0.2, par. 5.3.2, the LwM2M client
MUST send an “Update” operation to the LwM2M Server whenever the
lifetime parameter of the Server object changes the server). The same
applies for the object instances created/deleted. The changes in objects
seem to already be handled, but the lifetime was not.

Additionally, the "Update" message shall only contain these parameters
which changed since the last update (including objects). As it's
straightforward to determine if the liftime  changed but it's not easy
to tell if there were updates in the object instances, add an
additional parameter to the engine_trigger_update() function, indicating
that new object information shall be sent in the "Update" message.

Eventually add a proper error checking in `sm_send_registration` as the
function is reworked anyway.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

jukkar committed:
net: tcp2: Update seq when peer closes connection

If the peer ACKs data when it closes the connection, update
our sequence number accordinly. The connection would eventually
be terminated but this will avoid extra resends by the peer.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

ioannisg committed:
arch: arm: aarch32: cortex_m: improve documentation of z_arm_reset

We enhance the documentation of z_arm_reset, stressing that
the function may either be loaded by the processor coming
out of reset, or by another image, e.g. a bootloader. We
also specify what is required at minimum when executing the
reset function.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

pkral78 committed:
uart: native_posix: Add support for pts readiness

Option to pause writing to the pseudo terminal until it is ready to
receive data. Useful for pseudo terminal synchronization with other
host processes.

Signed-off-by: Pavel Král <pavel.kral@omsquare.com>

andrzej-kaczmarek committed:
Bluetooth: controller: Fix timings for non-1M PHY on advertising

We need to use actual phy wherever required to make sure radio delays
are calculated properly. This is especially important for LE Coded
since delays for that phy are significantly higher than on 1M and 2M.

Signed-off-by: Andrzej Kaczmarek <andrzej.kaczmarek@codecoup.pl>

nashif committed:
clang: remove include-fixed inclusion

This directory does not exist in clang/llvm, remove it.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
clang: remove cache handling

This is taking too long for every run and does not seem to work as
expected and wastes time instead of saving it. Remove for now..

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
llvm: add support for building x86_64-pc

set triplet for 64bit x86.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
clang: use LLVM_TOOLCHAIN_PATH instead of CLANG_ROOT_DIR

Be consistent with other toolchains.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

avisconti committed:
drivers/sensor: lsm6dsl: Fix build when irq_gpios is not in DT

In case of h/w setup with multiples device instances it is possible
that some of them wants to use triggers but not the others (no
interrupt line.

Since Kconfig configuration is the same way for all the instances
(CONFIG_LSM6DSL_TRIGGER=y), the driver behaves differently according
to how the device instance has been configured in the DT.
If irq-gpios is present, then the driver initialize the interrupt
part, else it skip irq init and returns ok, but data->gpio != NULL
xis checked in trigger_set() API.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

henrikbrixandersen committed:
samples: display: cfb_custom_font: add support for ssd1306fb

Add support for the Solomon SSD1306FB device to the CFB custom font
sample.

Convert sample to use DT_LABEL() for obtaining the device label instead
of hardcoding it.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

sdalu committed:
sensor_shell: added missing sensor channel.

Synchronized with list of channel defined in sensor.h

Signed-off-by: Stephane D'Alu <sdalu@sdalu.com>

nashif committed:
doc: fix showing latest version in sidebar

Minor fix for showing latest version in sidebar, no need for link here
that was added in recent commit.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

LingaoM committed:
Bluetooth: Mesh: Fix set friend_cred flag incorrectly

Only set friend_cred to true when friendship established and
use friend cred security material decryption successfully.

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

pabigot committed:
shell: fix declarations for external device arrays

The objects should be const-qualified for consistency with other uses,
including declarations in headers.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

lochej committed:
boards: nucleo_h743zi: enable ethernet support

Since STM32H7 series now have a working ethernet driver,
nucleo_h743zi ethernet can be enabled.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

nashif committed:
sanitycheck: count skipped tests due to overflow

Count tests that were skipped due to SRAM/RAM overdlow as skipped.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

scottwcpg committed:
boards: mec: mec15xxevb, mec1501modular: Update documentation

Update documention to refer to MEC152x specifications and
SPI image generator. MEC152x is the actual production SOC.
The only difference is the Boot-ROM loader SPI image layout.
Preserve the link to the old, MEC1501 SPI image generator.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

hollerdawg committed:
boards: frdm_k22f: Correct FXOS8700 i2c address

The address in NXP's documentation shows that 0x1c is the addressfor the
FXOS8700.

Signed-off-by: Ryan Holleran <rhollerar@gmail.com>

hollerdawg committed:
samples: sensor: fxos8700: Add frdm_k22f as a compatible platform

The frdm_k22f has the FX0S8700CQ placed. Add the frdm_k22f to the
FXOS8700  sample application.

Signed-off-by: Ryan Holleran <rhollerar@gmail.com>

nashif committed:
sanitycheck: handle overflow skips

skips due to ram/rom overflow were captured as failures. Fix this and
count them correctly as skips.

Fixes #29412

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

Commit:
test: atomic operation : add test case

Add tests with negative parameters to supplement black box tests.

Signed-off-by: Ying ming <mingx.ying@intel.com>

trond-snekvik committed:
Bluetooth: Mesh: Permit model walk from any model

Fixes bug where applications that disable model extensions end up in an
infinite loop, and adds support for walking model subtrees, as opposed
to forcing root to be unextended.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

Commit:
boards: arm: nucleo_h745zi_q: doc

add new modules description documentation

Signed-off-by: Alexander Kozhinov <AlexanderKozhinov@yandex.com>

nandojve committed:
drivers: wifi: eswifi: Drop WIFI_ESWIFI_NAME config

The WIFI_ESWIFI_NAME config would be uselful when there is no device
tree alternative.  The esWIFI driver already is on device tree and the
label property exists.  This remove WIFI_ESWIFI_NAME Kconfig variable
and switch to device tree equivalent.

Signed-off-by: Gerson Fernando Budke <gerson.budke@atl-electronics.com>

ehudmarvell committed:
net: dhcpv4: check dhcpv4 msg len is correct

check if dhcpv4 msg len is correct and drop it otherwise

Signed-off-by: Ehud Naim <ehudn@marvell.com>

ExtremeGTX committed:
drivers: wifi: rename WiFi kconfig prompt

use same naming convention for kconfig drivers entries

Signed-off-by: Mohamed ElShahawi <ExtremeGTX@hotmail.com>

ExtremeGTX committed:
net: ping: add warning if ping is not supported

Warn the user that "ping" is not supported in case of offloaded
driver is being used.

Signed-off-by: Mohamed ElShahawi <ExtremeGTX@hotmail.com>

DiwakarCC committed:
peci: driver: Ping command failing due to incorrect wait timeout

In current implementation, for Ping command, write data/payload is queued
in FIFO infinitely eventhough Ping command have write length of '0'.

This issue is addressed in this patch.

Signed-off-by: Diwakar-C <diwakar.c@intel.com>

DiwakarCC committed:
peci: driver: Ping cmd failing due to improper tx len and wait handling

In current implementation, for Ping command, write data/payload is
queued in FIFO infinitely eventhough Ping command have write length
of '0'.

This issue is addressed in this patch.

Signed-off-by: Diwakar C <diwakar.c@intel.com>

DiwakarCC committed:
peci: driver: Ping cmd failing due to improper tx len and wait handling

In current implementation, for Ping command, write data/payload is
queued in FIFO infinitely eventhough Ping command have write length
of '0'.

This issue is addressed in this patch.

Signed-off-by: Diwakar C <diwakar.c@intel.com>

erwango committed:
drivers/pinmux: stm32: Deprecate USB related macros

Following migration of all in-tree boards to device tree bindings
for USB pins configuration, deprecate USB related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/pinmux: stm32: Deprecate I2S related macros

Following migration of all in-tree boards to device tree bindings
for I2S pins configuration, deprecate I2S related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/pinmux: stm32: Deprecate SDMMC related macros

Following migration of all in-tree boards to device tree bindings
for SDMMC pins configuration, deprecate SDMMC related macros.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

DiwakarCC committed:
Merge branch 'peci_issue' of https://github.com/Diwakar-C/zephyr into peci_issue

DiwakarCC committed:
Merge branch 'peci_issue' of https://github.com/Diwakar-C/zephyr into peci_issue

DiwakarCC committed:
Merge branch 'peci_issue' of https://github.com/Diwakar-C/zephyr into peci_issue

erwango committed:
boards: stm32: Remove pinmux.c files

Following migration of pinctrl configuration from pinmux.c files
to device tree and deprecation of pinctrl defines, remove
pinmux.c files when possible.
Additionally remove the CMakeLists.txt files when it makes sense.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak committed:
west.yml: Pull in build fix for NXP IAP flash driver from HAL

The HAL was missing some new files for the IAP driver which causes build
failures.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

erwango committed:
boards: nucleo_g4*: Remove USB support

There is no user USB port available on these boards.
Remove support and remove files when no more needed.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

mniestroj committed:
drivers: wifi: esp: close stream socket on allocation failure

There is no reason to keep active stream socket when there was some data
loss. Mark such socket for closing and close it when all (so far)
received packets have been processed.

Signed-off-by: Marcin Niestroj <m.niestroj@grinn-global.com>

mniestroj committed:
drivers: wifi: esp: allow to configure RX net_pkt allocation timeout

There is now a hardcoded 100ms timeout on allocating new net_pkt for
received data. Move that configuration to Kconfig, so that value can be
tuned according to application needs.

Signed-off-by: Marcin Niestroj <m.niestroj@grinn-global.com>

mniestroj committed:
drivers: wifi: esp: increase default RX net_pkt allocation timeout

So far there was 100ms timeout on allocation of RX net_pkt. This is too
little for cases when lots of data are incoming on pretty
fast (e.g. 1Mbps) UART interface and application layer does not consume
received network packets fast enough. Up to now in such cases all
incoming data was processed from UART and there was no data loss when
utilizing hardware flow control. However there was high chance that data
processed from UART could not be passed further to network stack,
because of the 100ms net_pkt allocation timeout. This happens for
example when low priority application does not have enough time to run.

Increase default RX net_pkt allocation timeout from 100ms to 5s, so
there is much more time to process network packets. Such timeout should
not harm, because only a dedicated RX thread will be suspended for that
time, resulting in suspending UART traffic if hardware flow control is
supported.

Signed-off-by: Marcin Niestroj <m.niestroj@grinn-global.com>

albertofloyd commented on drivers/peci/peci_mchp_xec.c:
```diff
@@ -175,11 +170,13 @@ static int peci_xec_write(const struct device *dev, struct peci_msg *msg)
 #else
-	wait_timeout = tx_buf->len;
+	/* In worst case, overall timeout will be 1msec (100 * 10usec) */
+	u8_t wait_timeout_cnt = 100;
```
Need to update types uint8

franciscomunoz replied:
just fix the types everything else looks ok

trond-snekvik committed:
Bluetooth: Mesh: Account for ASZMIC in encrypt

The decision to set aszmic = 1 in the net_tx structure was made after
the variable was passed to the crypto context, creating a discrepancy
between the two when aszmic is 1.

Extracts transport encryption to a separate function.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

pfalcon committed:
MAINTAINERS: Remove myself from POSIX subsys maintainers.

To not keep the entry empty, move to collaborators.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

rlubos committed:
net: lwm2m: Add a callback to notify socket errors to engine users

Currently, when socket errors occur during receive, the LwM2M engine
restarts the state machine and registers again to the server. While this
works in simple use case (only RD client socket open), it's not a valid
approach when more sockets are open (FW update socket).

Fix this by introducing socket fault callback, which is registered by
the LwM2M engine users. This way, a proper socket owner is notified on
error and can pertake appropriate action.

For RD socket errors the behaviour remains the same - the state machine
is reset and the client registers again to the server. For FW update
socket, handle the error by reopening the socket and retransmitting the
last request. This allows to resume the download from the point the
error occured, w/o a need to start from scratch.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

Commit:
doc: update known-issues for Sphinx==3.3.0

Sphinx==3.3.0 slightly changed the warning messages, which causes fails
on filter-known-issues. Update to the new messages.

Signed-off-by: Fabio Utzig <fabio.utzig@nordicsemi.no>

Commit:
doc: enable documentation build for known-issues

Changing .known-issues/doc should always trigger a documentation build
so add it to the filters.

Signed-off-by: Fabio Utzig <fabio.utzig@nordicsemi.no>

mniestroj committed:
net: tls: fix tls_context leak in ztls_socket() error path

If there are enough tls_context objects in the system (configured by
CONFIG_NET_SOCKETS_TLS_MAX_CONTEXTS), but there are not enough file
descriptors (configured by CONFIG_POSIX_MAX_FDS) to create underneath
TCP/UDP socket, then TLS socket creation fails with leaked tls_context.

Call tls_release() in ztls_socket() error path whenever underneath
TCP/UDP socket creation fails.

Signed-off-by: Marcin Niestroj <m.niestroj@grinn-global.com>

galak committed:
ci: github: Update for deprecation of add-path

Github has deprecated add-path, so update the workflows that use it to
the new method of setting the PATH.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

pabigot committed:
doc: bump sphinx>=3.3.0

The filters were updated to match the 3.3.0 generated text, so update
the sphinx requirement to exclude versions that use different text.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

de-nordic committed:
west.yml: Update modules/fs/fatfs to latest

The FAT FS driver module has been updated to latest

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

mbolivar-nordic committed:
scripts: testedtlib: add more child-binding tests

The child_binding object should default to having a path and
compatible that matches the parent's. Mark it as xfail because the
compatible part is failing.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
scripts: edtlib: child binding compatibles match parents

Fixes: #29758

Commit 7165b77a81f35bd76fbbdc391f8c76f3d64d58a4 ("scripts: edtlib:
refactor for first class bindings") introduced a Binding class.

Its child_binding instance attribute has a compatible which can be
None. Adjust this behavior so the child Binding object has the same
compatible as the parent binding which ultimately has a compatible.

Without this, sanitycheck's expr_parser is doing some matching on
compatibles in child nodes that is producing unexpected results.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

albertofloyd commented:
Fixes https://github.com/zephyrproject-rtos/zephyr/pull/29708

edmont committed:
net: openthread: allow to configure platform info

This commit adds the option to configure the platform information
string of OpenThread.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

edmont committed:
net: openthread: fix maximum number of children range

This commit corrects the maximum allowed amount of children to
match Thread specification.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

galak committed:
mgmt/osdp: Fix build issue

When building the osdsp control_panel sample we get the following
compile error:

subsys/mgmt/osdp/src/osdp_cp.c:993:10: error: implicit declaration of
function 'osdp_cp_send_command_keyset'

Fix by adding ifdef protection around the call to
osdp_cp_send_command_keyset

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ehudmarvell committed:
logging: fix cmd_log_self_disable argc checker

according to the comment "all if no modules specified"
to support "all" state, set to 1 number of parameters.

Signed-off-by: Ehud Naim <ehudn@marvell.com>

mniestroj committed:
net: tls: use Maximum Fragment Length (MFL) extension by default

Call mbedtls_ssl_conf_max_frag_len() on created TLS context
configuration, so that Maximum Fragment Length (MFL) will be sent to
peer using RFC 6066 max_fragment_length extension. MFL value is
automatically chosen based on MBEDTLS_SSL_OUT_CONTENT_LEN and
MBEDTLS_SSL_IN_CONTENT_LEN mbed TLS macros.

This extension is mostly useful for TLS client side to tell TLS server
what is the maximum supported receive record length.

Signed-off-by: Marcin Niestroj <m.niestroj@grinn-global.com>

maxbachmann committed:
kernel: return error instead of misaligned k_thread object

k_object_alloc(K_OBJ_THREAD) returns a usable struct k_thread pointer.
This pointer is 4 byte aligned. On x86 and x86_64 struct _thread_arch
has a member which requires alignment. Since this is currently not
supported k_object_alloc(K_OBJ_THREAD) now returns an error instead of
a misaligned pointer.

Signed-off-by: Maximilian Bachmann <m.bachmann@acontis.com>

jukkar committed:
net: tcp2: Fix sending to 6lo based networks

Do not send the original pkt in 6lo based networks as in those
the IPv6 header is mangled and we would not be able to do any
resends of the original pkt. So for 6lo networks, clone the
pkt and send it to peer. The original pkt is kept in sent list
in case we need to resend to peer.

Fixes #29771

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
doc: net: Add info how to run IEEE 802.15.4 between two QEMUs

Add information how to run two QEMUs and have IEEE 802.15.4 link
layer over UART between them. This is useful if one needs to test
basic IEEE 802.15.4 functionality without a IEEE 802.15.4 radio.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

andrzej-kaczmarek committed:
Bluetooth: controller: Add common def for max AD size on legacy

This def can be used instead of hardcoding "31" everywhere.

Signed-off-by: Andrzej Kaczmarek <andrzej.kaczmarek@codecoup.pl>

jukkar committed:
doc: net: Add info how to capture IEEE 802.15.4 QEMU traffic

Add additional information how to capture IEEE 802.15.4 traffic
to a pcap file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

mmahadevan108 committed:
west.yml: Pull in build fix for NXP USB driver

The SDK USB driver was missing some defines that were
accidentally deleted when updating to SDK 2.8

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

Commit:
drivers: sensor: lsm6dsl: fixed irq_pin for interrupt handling

Obtaining irq_pin from dev leads to disabling interrupts
on unpredictable pin, as dev points to GPIO, not LSM6DSL.

Fixes #29721

Signed-off-by: Yurii Gubin <y.gubin@gmail.com>

galak committed:
drivers: remove unnecessary use of DEVICE_AND_API_INIT

For drivers that support CONFIG_DEVICE_POWER_MANAGEMENT there are some
cases that look like:

  #ifdef CONFIG_DEVICE_POWER_MANAGEMENT
    DEVICE_DEFINE()
  #else
    DEVICE_AND_API_INIT()
  #endif

There is no need to special case this as the pm_control_fn argument to
DEVICE_DEFINE will just be ignored in the
!CONFIG_DEVICE_POWER_MANAGEMENT case.  So we can cleanup the code a
little and remove the #else cases for the drivers that do this.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

jukkar committed:
net: tcp2: Update statistics

The amount of sent bytes and transmit errors should update
network statistics.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

trond-snekvik committed:
Bluetooth: controller: Silence unused variable warning with non-AE

Places definition of lll under the ADV_EXT #ifdef in lll_adv.c:isr_tx,
to silence build warnings for non-AE builds.

Follow-up from #29753.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

chrta committed:
drivers: ethernet: stm32: Add missing net_if_set_link_addr() call

When setting the MAC address, the ethernet driver has to call
net_if_set_link_addr() with the updated address. This was missing and is
added now.
See e.g.
https://github.com/zephyrproject-rtos/zephyr/pull/28874

Signed-off-by: Christian Taedcke <christian.taedcke@lemonbeat.com>

jukkar committed:
net: stats: Separate dropped TCP data segments and TCP packets

Track the number of dropped TCP data segments and number of dropped
TCP packets in network statistics. It is useful to see these
numbers separately.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: shell: Fix TCP statistics printing

Fix TCP statistics printing.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

Commit:
Test: add doxygen comment for test_pipe_thread2thread().

Add doxygen comments for details of test_pipe_thread2thread().

By the way, plan to do the same thing to all test cases
in test_pipe_contexts.c.

Signed-off-by: Steven Wang <steven.l.wang@linux.intel.com>

Commit:
doc: fix broken rST syntax in Kconfig help

Update usage of literal syntax in Kconfig file from markdown to
proper rST.

Signed-off-by: Fabio Utzig <fabio.utzig@nordicsemi.no>

Commit:
workflows: enable documentation build for Kconfig*

Trigger documentation builds when Kconfig files are changed.

Signed-off-by: Fabio Utzig <fabio.utzig@nordicsemi.no>

DiwakarCC committed:
peci: driver: Ping cmd fail due to improper tx len and wait handling

In current implementation, for Ping command, write data/payload is
queued in FIFO infinitely eventhough Ping command have write length
of '0'.

This issue is addressed in this patch.

Signed-off-by: Diwakar C <diwakar.c@intel.com>

DiwakarCC committed:
peci: driver: Ping cmd fail due to improper tx len and wait handling

In current implementation, for Ping command, write data/payload is
queued in FIFO infinitely eventhough Ping command have write length
of '0'.

This issue is addressed in this patch.

Signed-off-by: Diwakar C <diwakar.c@intel.com>

DiwakarCC commented:
Close
